### PR TITLE
feat: added Toml configuration

### PR DESCRIPTION
**Pull request by EnokiUN**

### CODE REVIEW

**Comment by EnokiUN** [comment]
Default `Eludris.toml` currently in my local eludris/eludris repo:

```toml
instance_name = "" # This is required, has to be over 0 characters long.

# You can uncomment the sections that you want and change their config values,
# this is just the defaults.

# With ratelimits, reset_after is the number of seconds between every bucket reset.

# [oprish.ratelimits]
# info = { reset_after = 5, limit = 2}
# message_create = { reset_after = 5, limit = 10}

# [pandemonium]
# ratelimit = { reset_after = 10, limit = 5}

# [effis]
# file_size = "100MB"
# ratelimit = { reset_after = 10, limit = 25, file_size_limit = "200MB"}
```

File: src/conf.rs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::conf::*;
+
+    #[test]
+    fn try_deserialize() {
+        let conf_str = r#"
```
Comment (ooliver1) [inline src/conf.rs:168]: Could this be a global constant so the indentation doesn't look so weird?

**Review by ooliver1 — DISMISSED** [review]

File: src/oprish/mod.rs [hunk 2]
```diff
@@ -73,9 +73,16 @@ impl<'r> FromRequest<'r> for ClientIP {
     type Error = Infallible;
 
     async fn from_request(req: &'r Request<'_>) -> Outcome<Self, Self::Error> {
-        if let Some(ip) = req.headers().get_one("X-Real-IP") {
-            Outcome::Success(ClientIP(IpAddr::from_str(ip).unwrap()))
-        } else if let Some(ip) = req.headers().get_one("CF-Connecting-IP") {
+        // Hey there future reader or probably oliver, in case you're wondering why these two lines
```
Comment (ooliver1) [inline src/oprish/mod.rs:76]: pogger

File: src/conf.rs (hunk 1 above)
Comment (EnokiUN) [reply src/conf.rs:168]: the point of it is it being in the cfg'd testing block which itself has one layer of indentation, it's actually not all that uncommon in rust as proven by [sqlx](https://github.com/launchbadge/sqlx#compile-time-verification)

**Pull request merged**